Cython/Compiler/Annotate.py

```diff
@@ -315,7 +315,7 @@ def annotate(match):
     br'(?P<py_macro_api>Py[A-Z][a-z]+_[A-Z][A-Z_]+)|'
     br'(?P<py_c_api>Py[A-Z][a-z]+_[A-Z][a-z][A-Za-z_]*)'
     br')(?=\()|'       # look-ahead to exclude subsequent '(' from replacement
-    br'(?P<error_goto>(?:(?<=;) *if [^;]* +)?__PYX_ERR\([^)]+\))'
+    br'(?P<error_goto>(?:(?<=;) *if \(((?!if).)*\) +)?__PYX_ERR\([^)]+\))'
```

Comment:
Instead of trying to match all text not having `;` character (`[^;]*`), I am trying to match `if (<if condition>)` statement. To ensure that `<if condition>` does not match another `if`-statement, I am not allowing `if condition` to contain `if` substring. Error handler is pretty static, so I think this is safe. In worst case we can extend `if` forbidden substring to e.g. `if (`.